pkg/gadgets/trace/dns/tracer/tracer.go

```diff
@@ -44,14 +44,40 @@ func NewTracer() (*Tracer, error) {
 		return nil, fmt.Errorf("failed to load asset: %w", err)
 	}
 
+	latencyCalc, err := newDNSLatencyCalculator()
+	if err != nil {
+		return nil, err
+	}
+
+	parseAndEnrichDNSEvent := func(rawSample []byte) (*types.Event, error) {
+		bpfEvent := (*dnsEventT)(unsafe.Pointer(&rawSample[0]))
+		if len(rawSample) < int(unsafe.Sizeof(*bpfEvent)) {
+			return nil, errors.New("invalid sample size")
+		}
+
+		event, err := bpfEventToDNSEvent(bpfEvent)
+		if err != nil {
+			return nil, err
+		}
+
+		// Derive latency from the request/response timestamps.
+		if event.Qr == types.DNSPktTypeQuery {
+			latencyCalc.storeDNSQueryTimestamp(bpfEvent.SaddrV6, bpfEvent.Id, bpfEvent.Timestamp)
```

Comment:
BTW, not related to this PR but `bpfEvent.SaddrV6` and `bpfEvent.DaddrV6`, should be just `bpfEvent.Saddr` and `bpfEvent.Daddr`. We can do it in another PR.

Reply:
It seems we have a mixed approach:

https://github.com/inspektor-gadget/inspektor-gadget/blob/91c1786200489a56acc530c202aad82acf5dd60a/pkg/gadgets/trace/dns/tracer/bpf/dns-common.h#L10-L17

https://github.com/inspektor-gadget/inspektor-gadget/blob/91c1786200489a56acc530c202aad82acf5dd60a/pkg/gadgets/trace/tcp/tracer/bpf/tcptracer.h#L22-L31

https://github.com/inspektor-gadget/inspektor-gadget/blob/91c1786200489a56acc530c202aad82acf5dd60a/pkg/gadgets/trace/tcpconnect/tracer/bpf/tcpconnect.h#L27-L34